ADR/00xx-component-references.md

```diff
@@ -0,0 +1,147 @@
+* When [integration-service] notices a build of a *referer* Component (one that declares that it embeds references to others), it does testing as normal, promotes to global candidate list as normal, and it promotes to dev env as normal, and it creates Releases as normal.
+* When [integration-service] notices a build of a Component that references no other components and no other components refer to it - then it proceeds like normal, runs tests, promotes, deploys, releases, etc.
+
+### Build-service
+
+* [build-service] will propagate the digest from one Component to another as a PR – following the declared `ComponentReferences`. This is a bit like renovatebot wrapped in a controller. The [build-service] acts when it sees that a build pipelinerun has completed successfully for a *referent* image.
+  * If this is the first time it has seen this *referent* updated in this way, it will file a new pull request on the *referer* supplying the pullspec and digest of the new referent, using the same branch name as was used for the *referent* PR.
+  * If a PR on that branch already exists, update that branch with an additional commit including the new *referent* digest.
+
+The [build-service] will also update the PR it filed when the PRs that triggered it are merged or updated. It may update the description and/or it may rebase on the main branch and/or it may issue /retest. Need to figure out what we want here.
+
+### “PR dependency”
+
+TODO - describe exactly how the user specifies that one PR depends on another. PaC will have to pass this through to the PipelineRun as an annotation. For now, just assume it is possible and defined.
+
+* When [integration-service] notices a build from a PR (PR#2) that declares it depends on another PR (PR#1), it will do testing but it will also:
+  * Construct the Snapshot for the test using the image from the triggering build pipeline of PR #2 as well as image found on the latest build pipeline associated with PR#1. This lets a user test PR #2 including unmerged content from PR #1.
+  * Post the test results back on PR#1 in addition to the results it would normally post to PR#2.
+  * Post a special followup check result on PR#2 that “fails” saying, “don’t merge this. It still depends on another.”
+* When [integration-service] notices a build from a PR (PR#2) that declares it depends on another PR (PR#1) but that other PR (PR#1) is merged, it will do testing and it will post a followup check result that “succeeds” saying “this is mergeable, all other PRs that it depends on are merged.”
+  * If PR #2 depends on two or more other PRs (not just 1), then [integration-service] should perform testing and post *distinct* followup check results on PR#2 reporting the merge status of all PRs it depends on. As they merge, those distinct check results turn from failing to passing. When all have passed, then PR#2 will appear as mergeable to the user.
```

Comment:
Imagine a situation where there are multiple PRs open on an embedded component's repository. There would have to be enough granularity present to open and track multiple distinct PRs on the embedding component's repository.